specs/Spec.HKDF.fsti

```diff
@@ -7,12 +7,12 @@ open Spec.Hash.Definitions
 val extract:
   a: hash_alg ->
   key: bytes{ Spec.HMAC.keysized a (Seq.length key) } ->
-  data: bytes{ Seq.length data + block_length a < max_input_length a } ->
-  Tot (lbytes (hash_length a))
+  data: bytes{ Seq.length data + block_length a <= Lib.IntTypes.max_size_t } ->
```

Comment:
This is a stricter bound than before, because of the length restriction on `Lib.Buffer`

Reply:
but conceivably we don't need to restrict the length in spec, we can just demand at run-time an `uint32` for the length and then show that any `uint32` satisfies `< max_input_length a`, right?

Reply:
The reason for this bound can be found in Spec.HMAC.fst (which Spec.HKDF is relying on).
Spec.HMAC uses Lib.Sequence.concat, which requires that the combined length of the two sequences is smaller than max_size_t (i.e. 2^32). 
An easy fix for this is to use FStar.Seq instead. The current limitation is to agree with _dev which uses Lib.Sequence.